charts/external-dns/templates/servicemonitor.yaml

```diff
@@ -22,32 +22,34 @@ spec:
     matchLabels:
       {{- include "external-dns.selectorLabels" . | nindent 6 }}
   endpoints:
-    - port: http
+    {{- range (list "http" "provider-metrics" }}
```

Comment:
I need to make the "provider-metrics" part of this conditional on there being a provider sidecar.